GMG/UI/Components/NavigationBar.swift

```diff
@@ -2,58 +2,53 @@
 
 import SwiftUI
 
-struct NavigationBar<Leading: View, Trailing: View>: View {
-    let title: String?
+struct NavigationBar<Leading: View, Center: View, Trailing: View>: View {
+    @Environment(\.colorScheme) private var colorScheme: ColorScheme
+
     let isBackButtonHidden: Bool
     let leading: Leading
+    let center: Center
     let trailing: Trailing
 
-    @Environment(\.colorScheme) private var colorScheme: ColorScheme
-
     init(
-        _ title: String? = nil,
         isBackButtonHidden: Bool = false,
         @ViewBuilder leading: () -> Leading,
+        @ViewBuilder center: () -> Center,
         @ViewBuilder trailing: () -> Trailing
     ) {
-        self.title = title
         self.isBackButtonHidden = isBackButtonHidden
         self.leading = leading()
+        self.center = center()
         self.trailing = trailing()
     }
 
     var body: some View {
-        ZStack {
-            HStack {
-                if !isBackButtonHidden {
-                    BackButton()
-                }
+        HStack {
+            if !isBackButtonHidden {
+                BackButton()
+            }
 
-                leading
+            leading
 
-                Spacer()
+            Spacer()
 
-                trailing
-            }
-            .buttonStyle(NavigationBarButtonStyle())
-
-            if let title {
-                Text(title)
-                    .font(Typography.WantedSansStd.R6)
-                    .foregroundStyle(
-                        colorScheme == .light
-                            ? Color.black1 : Color.white1
-                    )
-            }
+            center
+
+            Spacer()
+
+            trailing
         }
         .padding(Spacing.md)
+        .buttonStyle(NavigationBarButtonStyle())
     }
```

Comment:
HStack을 쓰면 trailing이 없으면 center가 오른쪽으로 치우치게 됩니다. ZStack으로 수정해주세요.

<p align="center">
  <img width="256" alt="NavigationBar" src="https://github.com/user-attachments/assets/590a2e3f-b642-43b5-8047-ab880c2cf610" />
</p>

```suggestion
    var body: some View {
        ZStack {
            HStack {
                if !isBackButtonHidden {
                    BackButton()
                }
                
                leading
                
                Spacer()
                
                trailing
            }
            
            center
        }
        .padding(Spacing.md)
        .buttonStyle(NavigationBarButtonStyle())
    }
```

Reply:
오 그렇네요! 중간이랑 양 옆의 레이아웃을 분리해서 넣는 방식이 다소 낯설긴한데 center를 정확하게 중간에 두는 좋은 방법이네요!